src/components/tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,266 @@
+  TooltipLayout
+} from "./utils/types";
+
+const screenDimensions = Dimensions.get("window");
+const INITIAL_COORDS: ChildrenCoords = {
+  x: 0,
+  y: 0,
+  width: 0,
+  height: 0
+};
+const ARROWS_BY_PLACEMENT: Record<
+  Placement,
+  JSXElementConstructor<{ color: string }>
+> = {
+  top: TopArrow,
+  bottom: BottomArrow,
+  left: LeftArrow,
+  right: RightArrow
+};
+
+type CommonProps = {
+  title: string;
+  content: string;
+  isVisible: boolean;
+  placement?: Placement;
+  displayInsets?: Partial<DisplayInsets>;
+  closeIconAccessibilityLabel: string;
+  childrenInteractionsEnabled?: boolean;
+  onClose: () => void;
+};
+type CloseWithTapOnBackground = {
+  allowCloseOnBackgroundTap: true;
+  backgroundAccessibilityLabel: string;
+};
+type CloseWithBackgroundTapDisabled = {
+  allowCloseOnBackgroundTap?: false;
+};
+type Props = CommonProps & (CloseWithTapOnBackground | CloseWithBackgroundTapDisabled);
+
+/**
+ * This component prompts a tooltip around its children in relation to the given `placement`.
+ * When `isVisible` is `true`
+ */
+export const Tooltip = ({
+  children,
+  title,
+  content,
+  placement: initialPlacement = "top",
+  closeIconAccessibilityLabel,
+  isVisible,
+  displayInsets = {},
+  allowCloseOnBackgroundTap,
+  childrenInteractionsEnabled,
+  onClose
+}: PropsWithChildren<Props>) => {
+  const insets = useSafeAreaInsets();
+  const [currentPlacement, setCurrentPlacement] =
+    useState<Placement>(initialPlacement);
+  const [childrenCoords, setChildrenCoords] = useState<ChildrenCoords>(INITIAL_COORDS);
+  const [tooltipLayout, setTooltipLayout] = useState<TooltipLayout>();
+  const childRef = useRef<View>(null);
+  const titleRef = useRef<View>(null);
+  const Arrow = useMemo(
+    () => ARROWS_BY_PLACEMENT[currentPlacement],
+    [currentPlacement]
+  );
+  const isChildrenMeasurementFinished =
+    every(childrenCoords, isDefined)
+    && some(childrenCoords, isNotZero);
+  const isTooltipMeasurementCompleted = isDefined(tooltipLayout);
+  const tooltipVisibility = { opacity: isTooltipMeasurementCompleted ? 1 : 0 };
+
+  /**
+   * This function sets the `Tooltip` children coordinates
+   */
+  const measureChildrenCoords = useCallback(() => {
+    if (childRef.current && typeof childRef.current.measure === "function") {
+      childRef.current.measure((_, __, width, height, px, py) => {
+        const coords = {
+          x: px,
+          y: py,
+          width,
+          height
+        };
+        if (every(coords, isDefined)) {
+          setChildrenCoords(coords);
+        }
+      });
+    }
+  }, []);
+
+  useEffect(() => {
+    if (isVisible) {
+      // A new measure is executed every time the `Tooltip` is visible
+      // This is required for use within ScrollView components.
+      setTimeout(measureChildrenCoords, 100);
+    } else {
+      setChildrenCoords(INITIAL_COORDS);
+      setCurrentPlacement(initialPlacement);
+    }
+  }, [isVisible, initialPlacement, measureChildrenCoords]);
```

Comment:
I think it's better to add clearTimeout, what do you think?
```suggestion
 useEffect(() => {
    // eslint-disable-next-line functional/no-let
    let timeoutId: string | number | NodeJS.Timeout | undefined;
    if (isVisible) {
      timeoutId = setTimeout(measureChildrenCoords, 100);
    } else {
      setChildrenCoords(INITIAL_COORDS);
      setCurrentPlacement(initialPlacement);
    }
    return () => clearTimeout(timeoutId);
  }, [isVisible, initialPlacement, measureChildrenCoords]);
```

Reply:
Addressed [here](https://github.com/pagopa/io-app-design-system/pull/351/commits/3320826cce913626aaeedbfa4201d8420d5b9942)